### PR TITLE
Socket creation under systemd and fix dependency issue

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@brainfunked , @r0h4n , @nthomas-redhat , @shtripat , @GowthamShanmugam , please review

**Comment by TimothyAsir** [comment]
tendrl-node-agent.socket should be added into MANIFEST.in

**Comment by TimothyAsirJeyasing** [comment]
@brainfunked , @r0h4n , @nthomas-redhat , @shtripat , @GowthamShanmugam , please review

**Comment by TimothyAsirJeyasing** [comment]
This patch fixes the issue: https://github.com/Tendrl/node-agent/issues/220

File: tendrl-node-agent.socket [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=tendrl node agent control socket
+DefaultDependencies=no
+Before=sockets.target
+
+[Socket]
+ListenSequentialPacket=/run/tendrl/message.sock
+SocketMode=0600
```
Comment (GowthamShanmugam) [inline tendrl-node-agent.socket]: @TimothyAsir why we need to give file message.sock also, it is created in mesage handler.

File: tendrl-node-agent.spec [hunk 2]
```diff
@@ -24,6 +24,7 @@ Requires: collectd
 Requires: python-jinja2
 Requires: tendrl-commons
 Requires: hwinfo 
+Requires: pytz
```
Comment (GowthamShanmugam) [inline tendrl-node-agent.spec]: @TimothyAsir is this really required? i never see this dependency when i test socket

File: tendrl-node-agent.socket (hunk 1 above)
Comment (TimothyAsirJeyasing) [inline tendrl-node-agent.socket]: Its not a problem, ListenSequentialPacket is the only option available for AF_UNIX sockets. If the value starts with a slash (/), it is read as file syste socket in the AF_UNIX socket family. please have a look at https://manned.org/systemd.socket/f4f82e36 for further details.

File: tendrl-node-agent.spec (hunk 2 above)
Comment (TimothyAsirJeyasing) [inline tendrl-node-agent.spec]: I received the following error during service start
  File "/usr/lib/python2.7/site-packages/tendrl/commons/message.py", line 6, in <module>
    from tendrl.commons.utils.time_utils import now
  File "/usr/lib/python2.7/site-packages/tendrl/commons/utils/time_utils.py", line 2, in <module>
    from pytz import utc
ImportError: No module named pytz

However it looks like this package "pytz" usage is newly introduced into commons
in tendrl/commons/utils/time_utils.py file. I will remove this here and will add into commons spec.

Comment (GowthamShanmugam) [inline tendrl-node-agent.spec]: @TimothyAsir understand this is dependency for time_utils

File: tendrl-node-agent.socket [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=tendrl node agent control socket
+DefaultDependencies=no
```
Comment (TimothyAsirJeyasing) [inline tendrl-node-agent.socket]: I Agree, I will do it

File: tendrl-node-agent.socket (hunk 1 above)
Comment (TimothyAsirJeyasing) [inline tendrl-node-agent.socket]: Sure, I will update.

File: tendrl-node-agent.spec [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Name: tendrl-node-agent
-Version: 1.2
+Version: 0.0
```
Comment (TimothyAsir) [inline tendrl-node-agent.spec]: oops, i changed the version for testing the rpm, forget to revert back. I will do revert this change.

**Comment by TimothyAsir** [comment]
@brainfunked , @r0h4n , @nthomas-redhat , @shtripat , please review

**Comment by TimothyAsir** [comment]
Yes, I verified the service start. Now the service is started without any issues.

**Comment by shtripat** [comment]
LGTM

**Comment by GowthamShanmugam** [comment]
@brainfunked @shtripat @brainfunked i have to change the socket creation and listen code to systemd?

File: tendrl-node-agent.socket [hunk 5]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=tendrl node agent control socket
+Before=sockets.target
+
+[Socket]
+ListenSequentialPacket=/run/tendrl/message.sock
```
Comment (r0h4n) [inline tendrl-node-agent.socket]: shouldnt this be /var/run/tendrl instead of /run/tendrl

**Comment by r0h4n** [comment]
@TimothyAsir Please update this PR

File: tendrl-node-agent.socket (hunk 5 above)
Comment (TimothyAsir) [inline tendrl-node-agent.socket]: Yes, it should be /var/run/tendrl to support older version. I will update this

**Comment by TimothyAsir** [comment]
@r0h4n , @nthomas-redhat , @shtripat , @anmolbabu , please review

**Pull request merged**